### ISSUE TITLE
Warnings generating chado_schema-1.3.yml (formerly foreign_key problems)

### ISSUE THREAD

**Issue by dsenalik** [tripal/tripal#1436]
# BUG/ERROR report

### Issue description
In Issue #1432 This comment https://github.com/tripal/tripal/issues/1432#issuecomment-1454840026 reports that the schema for project and at least one other table are missing the foreign keys.

#### Steps to reproduce
Look at ```tripal_chado/chado_schema/chado_schema-1.3.yml```
these lines, there are no foreign key definitions
```
   4888 project:
   4889   description: "Standard Chado flexible property table for projects."
   4890   fields:
   4891     project_id:
   4892       type: serial
   4893       not null: TRUE
   4894     name:
   4895       type: character varying
   4896       not null: TRUE
   4897       size: 255
   4898     description:
   4899       type: text
   4900       not null: FALSE
   4901   unique keys:
   4902     project_c1: name
   4903   primary key: project_id
   4904   referring_tables: assay_project, nd_experiment_project, project_analysis, project_contact, project_dbxref, project_feature, project_pub, project_relationship, project_stock, projectprop
```
The table right after ```project_contact:``` has foreign keys. I don't know how this file is generated, but I suspect it needs to be rebuilt.
For today's (20230304) docker the table does have foreign keys. In an earlier docker it did not have foreign keys.
```
sitedb=> \d project
                                           Table "chado.project"
   Column    |          Type          | Collation | Nullable |                   Default                   
-------------+------------------------+-----------+----------+---------------------------------------------
 project_id  | bigint                 |           | not null | nextval('project_project_id_seq'::regclass)
 name        | character varying(255) |           | not null | 
 description | text                   |           |          | 
Indexes:
    "project_pkey" PRIMARY KEY, btree (project_id)
    "project_c1" UNIQUE CONSTRAINT, btree (name)
Referenced by:
    TABLE "assay_project" CONSTRAINT "assay_project_project_id_fkey" FOREIGN KEY (project_id) REFERENCES project(project_id) DEFERRABLE INITIALLY DEFERRED
    TABLE "nd_experiment_project" CONSTRAINT "nd_experiment_project_project_id_fkey" FOREIGN KEY (project_id) REFERENCES project(project_id) ON DELETE CASCADE DEFERRABLE INITIALLY DEFERRED
    TABLE "project_analysis" CONSTRAINT "project_analysis_project_id_fkey" FOREIGN KEY (project_id) REFERENCES project(project_id) ON DELETE CASCADE DEFERRABLE INITIALLY DEFERRED
    TABLE "project_contact" CONSTRAINT "project_contact_project_id_fkey" FOREIGN KEY (project_id) REFERENCES project(project_id) ON DELETE CASCADE DEFERRABLE INITIALLY DEFERRED
    TABLE "project_dbxref" CONSTRAINT "project_dbxref_project_id_fkey" FOREIGN KEY (project_id) REFERENCES project(project_id) ON DELETE CASCADE DEFERRABLE INITIALLY DEFERRED
    TABLE "project_feature" CONSTRAINT "project_feature_project_id_fkey" FOREIGN KEY (project_id) REFERENCES project(project_id) ON DELETE CASCADE
    TABLE "project_pub" CONSTRAINT "project_pub_project_id_fkey" FOREIGN KEY (project_id) REFERENCES project(project_id) ON DELETE CASCADE DEFERRABLE INITIALLY DEFERRED
    TABLE "project_relationship" CONSTRAINT "project_relationship_object_project_id_fkey" FOREIGN KEY (object_project_id) REFERENCES project(project_id) ON DELETE CASCADE
    TABLE "project_relationship" CONSTRAINT "project_relationship_subject_project_id_fkey" FOREIGN KEY (subject_project_id) REFERENCES project(project_id) ON DELETE CASCADE
    TABLE "project_stock" CONSTRAINT "project_stock_project_id_fkey" FOREIGN KEY (project_id) REFERENCES project(project_id) ON DELETE CASCADE
    TABLE "projectprop" CONSTRAINT "projectprop_project_id_fkey" FOREIGN KEY (project_id) REFERENCES project(project_id) ON DELETE CASCADE
```

**Comment by dsenalik** [comment]
I found the program that generates the yml file, it is ```tripal_chado/chado_schema/generate_chado_schema_yaml.php```.
Running it in today's docker on the current 4.x branch
```drush php-script generate_chado_schema_yaml.php > test
 [warning] Undefined variable $arguments generate_chado_schema_yaml.php:15
 [warning] Trying to access array offset on value of type null generate_chado_schema_yaml.php:15
```
and foreign keys are still not included.
Warnings should be fixed because that part of the code doesn't do anything.

But the reason for no foreign keys is that they are defined in the linked table, not in  the project table.
An example of some foreign keys from ```default_schema-1.3.sql```
```
create table feature (
    ...
    foreign key (dbxref_id) references dbxref (dbxref_id) on delete set null INITIALLY DEFERRED,
    foreign key (organism_id) references organism (organism_id) on delete cascade INITIALLY DEFERRED,
    foreign key (type_id) references cvterm (cvterm_id) on delete cascade INITIALLY DEFERRED,
);
...
create table project (
    ...  (no foreign keys here!)
);
...
CREATE TABLE project_feature (
    ...
    FOREIGN KEY (feature_id) REFERENCES feature(feature_id) ON DELETE CASCADE,
    FOREIGN KEY (project_id) REFERENCES project(project_id) ON DELETE CASCADE
);
```

So the question becomes, should the yml file save the foreign keys from the other direction? I have gotten in too deep and beyond any semblance of knowing what I am doing now. 😵‍💫

**Comment by dsenalik** [comment]
&there4; There is not a problem with the yaml file, it is working as designed. It would be up to a field to check "from the other direction".
I will just update the title and create a pull request to deal with the two warnings in comment https://github.com/tripal/tripal/issues/1436#issuecomment-1454898188
```
[warning] Undefined variable $arguments generate_chado_schema_yaml.php:15
[warning] Trying to access array offset on value of type null generate_chado_schema_yaml.php:15
```

**Comment by dsenalik** [comment]
Fixed in merged PR #1438
